crates/rattler_conda_types/src/match_spec/mod.rs

```diff
@@ -173,6 +176,15 @@ impl Display for MatchSpec {
             None => write!(f, "*")?,
         }
 
+        match &self.extras {
+            Some(dependencies) => {
+                write!(f, "[")?;
+                write!(f, "{}", dependencies.iter().format(", "))?;
+                write!(f, "]")?;
+            }
+            None => {}
+        }
```

Comment:
THis will conflict if you also have for instance an `md5`. You should add `dependencies.iter...` as an elements into the `keys` vec below.